server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -233,53 +236,62 @@ func readScoreCard(ctx context.Context, env environment.Env, invocationID string
 }
 
 func (r *statsRecorder) Start() {
-	r.eg = errgroup.Group{}
 	ctx := context.Background()
 	for i := 0; i < numStatsRecorderWorkers; i++ {
-		r.eg.Go(func() error {
+		metrics.StatsRecorderWorkers.Inc()
+		go func() {
+			defer metrics.StatsRecorderWorkers.Dec()
 			for task := range r.tasks {
```

Comment:
If we had Go generics, I think we could write a nice little wrapper for creating and monitoring these types of worker pools. Some day :)